modules/dnn/include/opencv2/dnn/layer.hpp

```diff
@@ -66,6 +66,8 @@ class CV_EXPORTS LayerFactory
     //! Unregisters registered layer with specified type name. Thread-safe.
     static void unregisterLayer(const String &type);
 
+    static const std::map<String, int>& getRegisteredLayers() { return registeredTypes; }
```

Comment:
> `std::map<String, int>`

Why do we need `std::map` here?
Can we use simple `std::vector<std::string>` here?

Reply:
Removed this map, reused already existing [LayerFactory_Impl](https://github.com/opencv/opencv/blob/master/modules/dnn/src/dnn.cpp#L5313) map (moved it into private header), it has the same content.